lib/std/posix.zig

```diff
@@ -5653,16 +5623,43 @@ pub fn clock_getres(clock_id: clockid_t, res: *timespec) ClockGetTimeError!void
     }
 }
 
-pub const SchedGetAffinityError = error{PermissionDenied} || UnexpectedError;
+pub const SchedGetSetAffinityError = error{
+    SystemResources,
+    PermissionDenied,
+    ThreadNotFound,
+    OutOfMemory,
+} || UnexpectedError;
 
-pub fn sched_getaffinity(pid: pid_t) SchedGetAffinityError!cpu_set_t {
+pub fn sched_getaffinity(pid: pid_t) SchedGetSetAffinityError!cpu_set_t {
     var set: cpu_set_t = undefined;
-    switch (errno(system.sched_getaffinity(pid, @sizeOf(cpu_set_t), &set))) {
-        .SUCCESS => return set,
+    const size = @sizeOf(cpu_set_t);
+    const rc = system.sched_getaffinity(pid, size, &set);
+    const e = errno(rc);
+
+    if (e == .SUCCESS) {
+        if (!builtin.link_libc and rc < size) @memset(@as([*]u8, @ptrCast(&set))[rc..size], 0);
```

Comment:
The reason for the libc check is not obvious and could use a comment IMO

Reply:
Agree; added a comment.